File: data-pipelines/etl/etl-job.py
from pathlib import Path

def normalize(line: str) -> str:
  return line.strip()

def run_etl():
  src = Path("data/raw/raw-conversations.jsonl")
  dst = Path("data/datasets/cleaned/cleaned-conversations.jsonl")
  dst.parent.mkdir(parents=True, exist_ok=True)

  with src.open("r", encoding="utf-8") as fin, dst.open("w", encoding="utf-8") as fout:
    for line in fin:
      fout.write(normalize(line) + "\n")

if __name__ == "__main__":
  run_etl()

File: data-pipelines/etl/etl-job.ts
import { createReadStream, createWriteStream, mkdirSync } from 'fs';
import { join } from 'path';
import readline from 'readline';

export async function runEtl() {
  const src = join('data', 'datasets', 'raw', 'raw-conversations.jsonl');
  const dstDir = join('data', 'datasets', 'cleaned');
  const dst = join(dstDir, 'cleaned-conversations.jsonl');
  mkdirSync(dstDir, { recursive: true });

  const rl = readline.createInterface({ input: createReadStream(src) });
  const out = createWriteStream(dst);

  for await (const line of rl) {
    out.write(line.trim() + '\n');
  }
  out.end();
}
